kv/kvserver: TestReplicaRangefeedRetryErrors failed

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1813751&tab=buildLog) on [master@733563a521a9323f77411a609af1ea019073c6c9](https://github.com/cockroachdb/cockroach/commits/733563a521a9323f77411a609af1ea019073c6c9):

```
I200317 18:04:13.890654 396247 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED (3.10s)
        replica_rangefeed_test.go:455: change replicas of r2 failed: kv/kvserver/replica_raft.go:262: [n3,s3,r2/3:{a-/Max}]: received invalid ChangeReplicasTrigger ENTER_JOINT(r3 l3) REMOVE_REPLICA[(n3,s3):3VOTER_DEMOTING]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3VOTER_DEMOTING] next=4 to remove self (leaseholder)
=== RUN   TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT
I200317 18:04:19.543235 398446 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:40929" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200317 18:04:19.699918 398446 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200317 18:04:19.701564 398446 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:38547" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 18:04:19.710256 398689 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:40929
W200317 18:04:19.912514 398446 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 18:04:19.913860 398446 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:34583" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 18:04:19.917632 398998 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:40929
I200317 18:04:20.057695 398446 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 18:04:20.090605 398446 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=a77a5279] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 18:04:20.152373 398446 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 58ca93a2 at applied index 22
I200317 18:04:20.158213 398446 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 80, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200317 18:04:20.200503 399039 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200317 18:04:20.217128 399052 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200317 18:04:20.242244 399062 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=58ca93a2 index=22]
I200317 18:04:20.268613 399062 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=26ms ingestion=4@11ms id=58ca93a2 index=22]
I200317 18:04:20.289631 398446 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 18:04:20.302392 398446 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=0d659e72] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 18:04:20.336047 398446 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200317 18:04:20.385876 398446 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f34f3c30] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200317 18:04:20.502481 398446 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ba4f96f1 at applied index 27
I200317 18:04:20.510794 398446 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 87, log entries: 0, rate-limit: 8.0 MiB/sec, 0.07s
I200317 18:04:20.548128 399137 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200317 18:04:20.577384 399086 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=ba4f96f1 index=27]
I200317 18:04:20.603197 399086 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=25ms ingestion=4@20ms id=ba4f96f1 index=27]
I200317 18:04:20.625089 398446 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200317 18:04:20.850345 398446 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=a697ea94] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200317 18:04:21.109331 398446 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 18:04:21.615328 398687 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200317 18:04:22.015651 398815 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200317 18:04:22.098666 399506 util/stop/stopper.go:539  quiescing
I200317 18:04:22.099321 399504 util/stop/stopper.go:539  quiescing
I200317 18:04:22.100098 399505 util/stop/stopper.go:539  quiescing
W200317 18:04:22.135897 398970 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 18:04:22.186030 398970 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 18:04:22.192760 399111 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200317 18:04:22.194096 399110 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n2,s2):?
W200317 18:04:22.196433 399018 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200317 18:04:22.198072 399046 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: node unavailable; try another peer:
I200317 18:04:22.213809 399503 util/stop/stopper.go:539  quiescing
W200317 18:04:22.239671 399331 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200317 18:04:22.263941 399448 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
I200317 18:04:22.265192 399503 util/stop/stopper.go:539  quiescing
I200317 18:04:22.267026 399503 util/stop/stopper.go:539  quiescing
I200317 18:04:22.271033 399503 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (2.81s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584468261839242222 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1813893&tab=buildLog) on [master@bbe20c30457a9010c61c102b2feac2923c2b2c94](https://github.com/cockroachdb/cockroach/commits/bbe20c30457a9010c61c102b2feac2923c2b2c94):

```
=== RUN   TestReplicaRangefeedRetryErrors
--- FAIL: TestReplicaRangefeedRetryErrors (17.26s)
=== RUN   TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT
I200317 18:46:02.699085 403027 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33749" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200317 18:46:02.826228 403027 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200317 18:46:02.827402 403027 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:35113" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 18:46:02.833385 403110 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:33749
W200317 18:46:03.087519 403027 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 18:46:03.091357 403027 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:39425" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 18:46:03.112202 403542 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:33749
I200317 18:46:03.289848 403027 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 18:46:03.332496 403027 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=e27f6444] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 18:46:03.418322 403027 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 565aca3c at applied index 24
I200317 18:46:03.428847 403027 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 81, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200317 18:46:03.460042 403553 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=565aca3c index=24]
I200317 18:46:03.478274 403570 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200317 18:46:03.489285 403553 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=16ms ingestion=4@13ms id=565aca3c index=24]
I200317 18:46:03.503373 403027 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 18:46:03.568506 403027 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ea080843] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 18:46:03.652479 403027 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200317 18:46:03.798288 403027 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f8880ed6] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200317 18:46:03.849326 403027 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 6c6e0bea at applied index 32
I200317 18:46:03.863352 403027 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 89, log entries: 0, rate-limit: 8.0 MiB/sec, 0.02s
I200317 18:46:03.916007 403743 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=6c6e0bea index=32]
I200317 18:46:03.943345 403743 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=27ms ingestion=4@9ms id=6c6e0bea index=32]
I200317 18:46:03.961628 403027 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200317 18:46:04.097456 403027 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ffd76828] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200317 18:46:04.381413 403027 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 18:46:04.388619 403369 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200317 18:46:05.296906 404149 util/stop/stopper.go:539  quiescing
I200317 18:46:05.297795 404150 util/stop/stopper.go:539  quiescing
I200317 18:46:05.298473 404151 util/stop/stopper.go:539  quiescing
W200317 18:46:05.306210 403481 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200317 18:46:05.325302 403455 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
W200317 18:46:05.360120 403481 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200317 18:46:05.529888 404100 util/stop/stopper.go:539  quiescing
W200317 18:46:05.535142 403961 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200317 18:46:05.540167 404037 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200317 18:46:05.545599 403768 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200317 18:46:05.548172 403962 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200317 18:46:05.550006 403568 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200317 18:46:05.551850 403455 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200317 18:46:05.553406 403369 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200317 18:46:05.550189 403597 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200317 18:46:05.560829 404100 util/stop/stopper.go:539  quiescing
I200317 18:46:05.562737 404100 util/stop/stopper.go:539  quiescing
I200317 18:46:05.565443 404100 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (2.99s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584470764931814220 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814101&tab=buildLog) on [master@92b232044615a647d22fb220efcf53707cc0e4c7](https://github.com/cockroachdb/cockroach/commits/92b232044615a647d22fb220efcf53707cc0e4c7):

```
=== RUN   TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT
I200317 20:54:40.468081 411935 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:34877" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200317 20:54:40.556445 411935 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200317 20:54:40.557513 411935 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43815" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 20:54:40.561669 411971 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:34877
W200317 20:54:40.753386 411935 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 20:54:40.754690 411935 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:34233" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 20:54:40.762540 412301 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:34877
I200317 20:54:40.846850 411935 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 20:54:40.864508 411935 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=d6acaf7d] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 20:54:40.952186 411935 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 2326ae57 at applied index 21
I200317 20:54:40.974984 411935 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 79, log entries: 0, rate-limit: 8.0 MiB/sec, 0.04s
I200317 20:54:40.999890 412415 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=2326ae57 index=21]
I200317 20:54:41.022236 412415 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=22ms ingestion=4@3ms id=2326ae57 index=21]
I200317 20:54:41.050959 411935 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 20:54:41.112223 411935 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=09c5c4eb] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 20:54:41.172722 411935 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200317 20:54:41.224040 411935 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=304fcda7] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200317 20:54:41.356545 411935 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 8a3eab2f at applied index 28
I200317 20:54:41.363031 411935 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 87, log entries: 0, rate-limit: 8.0 MiB/sec, 0.06s
I200317 20:54:41.409777 412586 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=8a3eab2f index=28]
I200317 20:54:41.424500 412602 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200317 20:54:41.427897 412586 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=17ms ingestion=4@9ms id=8a3eab2f index=28]
I200317 20:54:41.462777 411935 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200317 20:54:41.772869 411935 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=3dcd156d] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200317 20:54:42.177819 411935 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 20:54:42.193440 412111 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200317 20:54:42.916867 413034 util/stop/stopper.go:539  quiescing
I200317 20:54:42.918034 413033 util/stop/stopper.go:539  quiescing
I200317 20:54:42.921785 413035 util/stop/stopper.go:539  quiescing
W200317 20:54:42.936617 412072 kv/kvserver/store.go:1541  [s1,r1/1:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 20:54:42.979624 412389 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200317 20:54:42.988788 412291 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
W200317 20:54:43.024389 412211 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 20:54:43.038376 412389 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 20:54:43.080133 412420 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200317 20:54:43.086618 412487 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200317 20:54:43.113665 413032 util/stop/stopper.go:539  quiescing
W200317 20:54:43.118851 412428 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200317 20:54:43.119837 412848 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200317 20:54:43.126592 412843 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200317 20:54:43.127332 412492 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200317 20:54:43.128190 412111 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200317 20:54:43.128587 412291 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200317 20:54:43.129968 412835 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200317 20:54:43.139933 413032 util/stop/stopper.go:539  quiescing
I200317 20:54:43.142647 413032 util/stop/stopper.go:539  quiescing
I200317 20:54:43.146515 413032 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (2.79s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584478482741517764 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814167&tab=buildLog) on [master@33ef1baf5a3d423b7d13c16c5d31166853a388d0](https://github.com/cockroachdb/cockroach/commits/33ef1baf5a3d423b7d13c16c5d31166853a388d0):

```
=== RUN   TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED
I200317 21:45:21.315312 402946 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:41321" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200317 21:45:21.441648 402946 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200317 21:45:21.443443 402946 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:36573" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 21:45:21.445951 403016 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:41321
W200317 21:45:21.617343 402946 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 21:45:21.618563 402946 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:45247" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 21:45:21.623578 403026 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:41321
I200317 21:45:21.743463 402946 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 21:45:21.771419 402946 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=dcad47b0] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 21:45:21.825651 402946 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 3c5d38f5 at applied index 21
I200317 21:45:21.860622 402946 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 79, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200317 21:45:21.887962 403469 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=3c5d38f5 index=21]
I200317 21:45:21.906648 403469 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=18ms ingestion=4@8ms id=3c5d38f5 index=21]
I200317 21:45:21.917812 403496 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200317 21:45:21.967690 402946 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 21:45:22.026704 402946 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=e6134ba2] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 21:45:22.046180 402946 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200317 21:45:22.097208 402946 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=dd114061] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200317 21:45:22.276817 402946 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ad82e1d5 at applied index 27
I200317 21:45:22.314581 402946 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 87, log entries: 0, rate-limit: 8.0 MiB/sec, 0.09s
I200317 21:45:22.370033 403607 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=ad82e1d5 index=27]
I200317 21:45:22.432847 403607 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=62ms ingestion=4@54ms id=ad82e1d5 index=27]
I200317 21:45:22.442000 403671 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200317 21:45:22.449303 402946 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200317 21:45:22.615221 402946 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=50dfa87d] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200317 21:45:22.969426 402946 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 21:45:22.997229 403132 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200317 21:45:23.766640 403426 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200317 21:45:24.011189 404036 util/stop/stopper.go:539  quiescing
I200317 21:45:24.011915 403954 util/stop/stopper.go:539  quiescing
I200317 21:45:24.012489 403955 util/stop/stopper.go:539  quiescing
W200317 21:45:24.015184 403101 kv/kvserver/store.go:1541  [s1,r1/1:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 21:45:24.072650 403412 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 21:45:24.132071 403412 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 21:45:24.148645 403513 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200317 21:45:24.150902 403540 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200317 21:45:24.230928 403953 util/stop/stopper.go:539  quiescing
W200317 21:45:24.252672 403794 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200317 21:45:24.252961 403793 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200317 21:45:24.253611 403852 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200317 21:45:24.254397 403493 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200317 21:45:24.259095 403583 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200317 21:45:24.260224 403132 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200317 21:45:24.260932 403426 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 21:45:24.264078 403953 util/stop/stopper.go:539  quiescing
I200317 21:45:24.266040 403953 util/stop/stopper.go:539  quiescing
I200317 21:45:24.268477 403953 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED (3.06s)
        replica_rangefeed_test.go:452: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584481523611296563 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1814326&tab=buildLog) on [master@33ef1baf5a3d423b7d13c16c5d31166853a388d0](https://github.com/cockroachdb/cockroach/commits/33ef1baf5a3d423b7d13c16c5d31166853a388d0):

```
I200317 23:55:13.376294 406916 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:43457" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 23:55:13.403700 407143 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:42469
W200317 23:55:13.641462 406916 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200317 23:55:13.642884 406916 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:43585" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200317 23:55:13.645875 407366 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:42469
I200317 23:55:13.761380 406916 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200317 23:55:13.778608 406916 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ecba1b77] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200317 23:55:13.816125 406916 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ed3b106f at applied index 22
I200317 23:55:13.818988 406916 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 80, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200317 23:55:13.838924 407452 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=ed3b106f index=22]
I200317 23:55:13.851059 407452 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=12ms ingestion=4@4ms id=ed3b106f index=22]
I200317 23:55:13.915900 406916 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200317 23:55:14.075967 406916 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=fc038105] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200317 23:55:14.170639 406916 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200317 23:55:14.401613 406916 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=42364d51] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200317 23:55:14.493884 406916 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot c00684ef at applied index 32
I200317 23:55:14.530347 406916 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 89, log entries: 0, rate-limit: 8.0 MiB/sec, 0.08s
I200317 23:55:14.571063 407701 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=c00684ef index=32]
I200317 23:55:14.598281 407659 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200317 23:55:14.631960 407701 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=60ms ingestion=4@30ms id=c00684ef index=32]
I200317 23:55:14.716888 406916 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200317 23:55:14.976852 406916 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7cdc1c36] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200317 23:55:15.272240 406916 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200317 23:55:15.390591 407261 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200317 23:55:15.730147 406975 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200317 23:55:15.903982 407296 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200317 23:55:16.450442 408122 util/stop/stopper.go:539  quiescing
I200317 23:55:16.451171 408120 util/stop/stopper.go:539  quiescing
I200317 23:55:16.452448 408121 util/stop/stopper.go:539  quiescing
W200317 23:55:16.497825 407249 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 23:55:16.502212 407068 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200317 23:55:16.502522 407068 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200317 23:55:16.503851 407068 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200317 23:55:16.504192 407068 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200317 23:55:16.532298 407361 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200317 23:55:16.533824 407409 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200317 23:55:16.534837 407409 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200317 23:55:16.542113 407394 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200317 23:55:16.643414 408119 util/stop/stopper.go:539  quiescing
W200317 23:55:16.647487 407261 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200317 23:55:16.647865 407976 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200317 23:55:16.651153 407944 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200317 23:55:16.653043 407945 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200317 23:55:16.654448 407296 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200317 23:55:16.654794 407374 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
I200317 23:55:16.662529 408119 util/stop/stopper.go:539  quiescing
I200317 23:55:16.664190 408119 util/stop/stopper.go:539  quiescing
I200317 23:55:16.667331 408119 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED (3.53s)
        replica_rangefeed_test.go:452: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584489315957629888 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815658&tab=buildLog) on [master@afd27c255065909f357f37f7c38309592e5b0da6](https://github.com/cockroachdb/cockroach/commits/afd27c255065909f357f37f7c38309592e5b0da6):

```
I200318 15:19:36.249952 408541 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:44875
W200318 15:19:36.440641 408300 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200318 15:19:36.442510 408300 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:38473" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 15:19:36.458683 408619 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:44875
I200318 15:19:36.565877 408300 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 15:19:36.596243 408300 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=2150494f] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 15:19:36.694781 408300 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot fb417fa8 at applied index 22
I200318 15:19:36.702908 408300 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 79, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200318 15:19:36.734410 408827 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=fb417fa8 index=22]
I200318 15:19:36.744470 408874 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 15:19:36.767276 408827 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=32ms ingestion=4@20ms id=fb417fa8 index=22]
I200318 15:19:36.837480 408300 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 15:19:36.896749 408300 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=8b83456a] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 15:19:36.981906 408300 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200318 15:19:37.146472 408300 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=3534a848] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200318 15:19:37.194490 408300 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 3b624d99 at applied index 32
I200318 15:19:37.241490 408300 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 89, log entries: 0, rate-limit: 8.0 MiB/sec, 0.06s
I200318 15:19:37.269014 408899 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=3b624d99 index=32]
I200318 15:19:37.287100 408899 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=18ms ingestion=4@3ms id=3b624d99 index=32]
I200318 15:19:37.381813 408300 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200318 15:19:37.520661 408300 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=5d6b2883] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200318 15:19:37.777112 408599 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200318 15:19:37.787776 408300 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 15:19:38.007076 408748 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200318 15:19:38.662822 409236 util/stop/stopper.go:539  quiescing
I200318 15:19:38.663683 409234 util/stop/stopper.go:539  quiescing
I200318 15:19:38.665524 409235 util/stop/stopper.go:539  quiescing
W200318 15:19:38.668503 408440 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200318 15:19:38.670520 408440 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200318 15:19:38.681924 408440 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200318 15:19:38.682346 408440 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200318 15:19:38.685140 408732 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200318 15:19:38.690452 408727 kv/kvserver/queue.go:578  [s3,r2/3:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200318 15:19:38.712758 408560 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 15:19:38.732162 408732 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 15:19:38.776933 408856 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 15:19:38.790520 408871 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200318 15:19:38.840112 409233 util/stop/stopper.go:539  quiescing
W200318 15:19:38.850854 409198 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: rpc error: code = Unavailable desc = transport is closing:
W200318 15:19:38.851576 409168 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 15:19:38.852452 409167 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 15:19:38.856497 408658 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 15:19:38.857815 408942 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200318 15:19:38.858666 408599 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200318 15:19:38.859484 408748 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200318 15:19:38.861339 409233 util/stop/stopper.go:539  quiescing
I200318 15:19:38.863491 409233 util/stop/stopper.go:539  quiescing
I200318 15:19:38.866562 409233 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (2.85s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584544778393826151 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815782&tab=buildLog) on [master@c4ed84bf1e3797e753ebbabe158de7c13ca35adc](https://github.com/cockroachdb/cockroach/commits/c4ed84bf1e3797e753ebbabe158de7c13ca35adc):

```
I200318 16:01:51.180112 404113 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:33103
W200318 16:01:51.384775 404008 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200318 16:01:51.385846 404008 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:42149" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 16:01:51.389533 404284 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:33103
I200318 16:01:51.457944 404008 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 16:01:51.490498 404008 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7d868f8f] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 16:01:51.583633 404008 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot ae2b1b16 at applied index 21
I200318 16:01:51.638873 404008 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 79, log entries: 0, rate-limit: 8.0 MiB/sec, 0.11s
I200318 16:01:51.669269 404520 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 16:01:51.678485 404292 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=ae2b1b16 index=21]
I200318 16:01:51.702735 404292 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=24ms ingestion=4@13ms id=ae2b1b16 index=21]
I200318 16:01:51.717864 404008 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 16:01:51.792460 404008 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=bbf87047] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 16:01:51.877123 404008 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200318 16:01:51.972739 404008 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=9f81be37] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200318 16:01:52.101255 404008 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 4fe03449 at applied index 30
I200318 16:01:52.104764 404008 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 87, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200318 16:01:52.133856 404630 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=4fe03449 index=30]
I200318 16:01:52.195382 404630 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=60ms ingestion=4@28ms id=4fe03449 index=30]
I200318 16:01:52.246233 404008 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200318 16:01:52.319717 404008 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7bf39c68] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200318 16:01:52.720580 404008 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 16:01:52.857132 404112 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200318 16:01:53.032870 404394 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
I200318 16:01:53.592791 404889 util/stop/stopper.go:539  quiescing
I200318 16:01:53.593063 404890 util/stop/stopper.go:539  quiescing
I200318 16:01:53.593201 404891 util/stop/stopper.go:539  quiescing
W200318 16:01:53.599032 404114 kv/txn.go:590  [liveness-hb] failure aborting transaction: failed to send RPC: sending to all 3 replicas failed; last error: (err: node unavailable; try another peer) <nil>; abort caused by: result is ambiguous (server shutdown)
I200318 16:01:53.599430 404114 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200318 16:01:53.627601 404381 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 16:01:53.639203 404140 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 16:01:53.666569 404114 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200318 16:01:53.666991 404114 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200318 16:01:53.687045 404381 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 16:01:53.696941 404140 kv/kvserver/store.go:1541  [s2,r1/2:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 16:01:53.726424 404707 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W200318 16:01:53.763549 404784 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200318 16:01:53.765798 404518 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W200318 16:01:53.806209 404501 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200318 16:01:53.844801 404888 util/stop/stopper.go:539  quiescing
W200318 16:01:53.857248 404397 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
W200318 16:01:53.857965 404394 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200318 16:01:53.859342 404778 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 16:01:53.875149 404510 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 16:01:53.875845 404112 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 16:01:53.892863 404888 util/stop/stopper.go:539  quiescing
I200318 16:01:53.895311 404888 util/stop/stopper.go:539  quiescing
I200318 16:01:53.898973 404888 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED (3.00s)
        replica_rangefeed_test.go:452: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584547313286886305 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1815929&tab=buildLog) on [master@b5f030223fbcf22e806c48a3c46e74a73a54f50f](https://github.com/cockroachdb/cockroach/commits/b5f030223fbcf22e806c48a3c46e74a73a54f50f):

```
I200318 17:23:58.349735 402866 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200318 17:23:58.564938 402866 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=6f22953e] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200318 17:23:58.774135 402866 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 6c7d3d6e at applied index 36
I200318 17:23:58.780290 402866 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 90, log entries: 0, rate-limit: 8.0 MiB/sec, 0.06s
I200318 17:23:58.817937 403521 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200318 17:23:58.850540 403549 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=6c7d3d6e index=36]
I200318 17:23:58.892981 403549 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=39ms ingestion=4@24ms id=6c7d3d6e index=36]
I200318 17:23:58.902919 402704 gossip/gossip.go:1527  [n1] node has connected to cluster via gossip
I200318 17:23:58.926323 403560 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200318 17:23:59.038837 402866 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200318 17:23:59.138160 403172 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200318 17:23:59.193034 402866 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f432c734] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200318 17:23:59.555442 402866 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 17:23:59.579696 403089 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
W200318 17:24:00.039780 403090 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms:
  - context deadline exceeded
I200318 17:24:00.189431 403173 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (context deadline exceeded)
W200318 17:24:00.192307 403173 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms
I200318 17:24:00.741166 403173 kv/kvserver/node_liveness.go:461  [liveness-hb] heartbeat failed on epoch increment; retrying
I200318 17:24:01.008666 403972 util/stop/stopper.go:539  quiescing
I200318 17:24:01.011471 403922 util/stop/stopper.go:539  quiescing
I200318 17:24:01.017766 403923 util/stop/stopper.go:539  quiescing
I200318 17:24:01.037966 403787 kv/kvserver/queue.go:1189  [replicate] purgatory is now empty
W200318 17:24:01.043006 403090 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200318 17:24:01.043439 403090 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200318 17:24:01.046626 403013 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200318 17:24:01.046956 403013 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200318 17:24:01.049323 403298 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 17:24:01.049641 402983 kv/kvserver/store.go:1541  [s1,r1/1:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 17:24:01.057903 403090 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200318 17:24:01.068269 403090 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
I200318 17:24:01.069062 403919 kv/kvserver/queue.go:1189  [replicate] purgatory is now empty
W200318 17:24:01.121536 403298 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 17:24:01.230347 403298 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 17:24:01.235646 403374 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 17:24:01.240946 403374 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 17:24:01.246592 403358 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200318 17:24:01.326737 403921 util/stop/stopper.go:539  quiescing
W200318 17:24:01.333031 403339 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 17:24:01.342233 403745 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 17:24:01.342481 403089 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200318 17:24:01.344784 403746 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 17:24:01.346112 403172 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200318 17:24:01.346719 403441 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200318 17:24:01.347782 403803 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
I200318 17:24:01.353620 403921 util/stop/stopper.go:539  quiescing
I200318 17:24:01.355644 403921 util/stop/stopper.go:539  quiescing
I200318 17:24:01.358502 403921 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_REPLICA_REMOVED (4.26s)
        replica_rangefeed_test.go:452: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584552240426032467 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816088&tab=buildLog) on [master@aaa604038de76a0dbbecf913ba8a6808f9ad6eae](https://github.com/cockroachdb/cockroach/commits/aaa604038de76a0dbbecf913ba8a6808f9ad6eae):

```
=== RUN   TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT
I200318 19:04:15.336637 402124 gossip/gossip.go:395  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:40217" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
W200318 19:04:15.476926 402124 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
I200318 19:04:15.478115 402124 gossip/gossip.go:395  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:45527" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 19:04:15.482047 402192 gossip/client.go:124  [n2] started gossip client to 127.0.0.1:40217
W200318 19:04:15.693990 402124 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
I200318 19:04:15.695376 402124 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:38681" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 19:04:15.703012 402491 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:40217
I200318 19:04:15.742313 402124 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 19:04:15.761251 402124 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=1eb54e74] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 19:04:15.790635 402124 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot cf20b3f1 at applied index 21
I200318 19:04:15.809861 402124 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 79, log entries: 0, rate-limit: 8.0 MiB/sec, 0.03s
I200318 19:04:15.860451 402662 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=cf20b3f1 index=21]
I200318 19:04:15.872586 402468 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 19:04:15.881488 402662 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=20ms ingestion=4@15ms id=cf20b3f1 index=21]
I200318 19:04:15.919955 402124 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 19:04:15.959193 402124 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=895e6ff4] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 19:04:16.063184 402124 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200318 19:04:16.128008 402124 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=061502f4] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200318 19:04:16.234201 402124 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 9233d37a at applied index 27
I200318 19:04:16.239774 402124 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 87, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200318 19:04:16.336606 402737 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=9233d37a index=27]
I200318 19:04:16.361446 402737 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=22ms ingestion=4@16ms id=9233d37a index=27]
I200318 19:04:16.376870 402767 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n3,s3):3LEARNER
I200318 19:04:16.444448 402124 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200318 19:04:16.542749 402124 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=f035b801] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200318 19:04:16.870988 402124 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 19:04:17.699328 402456 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200318 19:04:17.754865 403092 util/stop/stopper.go:539  quiescing
I200318 19:04:17.759523 402529 kv/kvserver/queue.go:578  [s3,r2/3:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
I200318 19:04:17.766300 403091 util/stop/stopper.go:539  quiescing
I200318 19:04:17.767610 403090 util/stop/stopper.go:539  quiescing
W200318 19:04:17.773209 402263 kv/kvserver/store.go:1541  [s1,r1/1:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 19:04:17.777520 402193 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200318 19:04:17.778192 402193 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200318 19:04:17.832035 402639 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 19:04:17.836531 402639 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 19:04:17.840462 402621 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
I200318 19:04:17.933045 403089 util/stop/stopper.go:539  quiescing
W200318 19:04:17.941875 402914 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 19:04:17.942790 402980 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 19:04:17.946442 402915 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 19:04:17.947144 402456 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200318 19:04:17.947911 402495 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Canceled desc = grpc: the client connection is closing:
W200318 19:04:17.951598 402780 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: rpc error: code = Canceled desc = grpc: the client connection is closing:
I200318 19:04:17.952662 403089 util/stop/stopper.go:539  quiescing
I200318 19:04:17.954395 403089 util/stop/stopper.go:539  quiescing
I200318 19:04:17.957235 403089 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (2.74s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584558257559246542 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(kv/kvserver).TestReplicaRangefeedRetryErrors failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1816324&tab=buildLog) on [master@37f2a6964760b5d72bef655342628e31601b8978](https://github.com/cockroachdb/cockroach/commits/37f2a6964760b5d72bef655342628e31601b8978):

```
I200318 20:43:00.039464 400152 gossip/gossip.go:395  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:44565" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 cluster_name:"" sql_address:<network_field:"" address_field:"" > 
I200318 20:43:00.064787 400587 gossip/client.go:124  [n3] started gossip client to 127.0.0.1:43473
I200318 20:43:00.256608 400152 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I200318 20:43:00.314693 400152 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=ece6bd67] proposing SIMPLE(l2) ADD_REPLICA[(n2,s2):2LEARNER]: after=[(n1,s1):1 (n2,s2):2LEARNER] next=3
I200318 20:43:00.374873 400152 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 11e8d6d0 at applied index 26
I200318 20:43:00.401285 400152 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):2LEARNER: kv pairs: 81, log entries: 0, rate-limit: 8.0 MiB/sec, 0.05s
I200318 20:43:00.430223 400638 kv/kvserver/replica_raftstorage.go:780  [s2,r1/2:{-}] applying LEARNER snapshot [id=11e8d6d0 index=26]
I200318 20:43:00.443949 400638 kv/kvserver/replica_raftstorage.go:801  [s2,r1/2:/M{in-ax}] applied LEARNER snapshot [total=13ms ingestion=4@6ms id=11e8d6d0 index=26]
I200318 20:43:00.465508 400730 kv/kvserver/raft_snapshot_queue.go:126  [raftsnapshot,s1,r1/1:/M{in-ax}] skipping snapshot; replica is likely a learner in the process of being added: (n2,s2):2LEARNER
I200318 20:43:00.493999 400152 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n2,s2):2] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2LEARNER, next=3, gen=1]
I200318 20:43:00.678262 400152 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=db762c88] proposing SIMPLE(v2) ADD_REPLICA[(n2,s2):2]: after=[(n1,s1):1 (n2,s2):2] next=3
I200318 20:43:00.763958 400152 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3LEARNER] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=2]
I200318 20:43:00.826837 400152 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=1d0e7418] proposing SIMPLE(l3) ADD_REPLICA[(n3,s3):3LEARNER]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3LEARNER] next=4
I200318 20:43:00.979090 400152 kv/kvserver/store_snapshot.go:975  [s1,r1/1:/M{in-ax}] sending LEARNER snapshot 3ba8631f at applied index 35
I200318 20:43:01.029180 400152 kv/kvserver/store_snapshot.go:1018  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):3LEARNER: kv pairs: 90, log entries: 0, rate-limit: 8.0 MiB/sec, 0.09s
I200318 20:43:01.045849 400871 kv/kvserver/replica_raftstorage.go:780  [s3,r1/3:{-}] applying LEARNER snapshot [id=3ba8631f index=35]
I200318 20:43:01.071032 400871 kv/kvserver/replica_raftstorage.go:801  [s3,r1/3:/M{in-ax}] applied LEARNER snapshot [total=18ms ingestion=4@8ms id=3ba8631f index=35]
I200318 20:43:01.124529 400152 kv/kvserver/replica_command.go:1602  [s1,r1/1:/M{in-ax}] change replicas (add [(n3,s3):3] remove []): existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, (n3,s3):3LEARNER, next=4, gen=3]
I200318 20:43:01.407518 400152 kv/kvserver/replica_raft.go:241  [s1,r1/1:/M{in-ax},txn=7d79f9d9] proposing SIMPLE(v3) ADD_REPLICA[(n3,s3):3]: after=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I200318 20:43:01.790122 400457 gossip/gossip.go:1527  [n2] node has connected to cluster via gossip
I200318 20:43:01.858240 400152 kv/kvserver/replica_command.go:397  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "a" [r2] (manual)
I200318 20:43:02.000949 400586 gossip/gossip.go:1527  [n3] node has connected to cluster via gossip
W200318 20:43:02.436016 400258 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: operation "node liveness heartbeat" timed out after 450ms:
  - context deadline exceeded
I200318 20:43:03.121900 401286 util/stop/stopper.go:539  quiescing
I200318 20:43:03.127233 401287 util/stop/stopper.go:539  quiescing
I200318 20:43:03.127978 401288 util/stop/stopper.go:539  quiescing
W200318 20:43:03.134651 400570 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200318 20:43:03.164812 400519 kv/kvserver/queue.go:578  [s3,r2/3:{a-/Max}] rate limited in MaybeAdd (replicate): node unavailable; try another peer
W200318 20:43:03.165821 400258 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: result is ambiguous (server shutdown)
I200318 20:43:03.167977 400258 kv/kvserver/node_liveness.go:804  [liveness-hb] retrying liveness update after kvserver.errRetryLiveness: result is ambiguous (server shutdown)
W200318 20:43:03.180594 400258 kv/txn.go:590  [liveness-hb] failure aborting transaction: node unavailable; try another peer; abort caused by: node unavailable; try another peer
W200318 20:43:03.181190 400258 kv/kvserver/node_liveness.go:471  [liveness-hb] failed node liveness heartbeat: node unavailable; try another peer
W200318 20:43:03.192041 400570 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
W200318 20:43:03.271963 400670 kv/kvserver/raft_transport.go:294  unable to accept Raft message from (n3,s3):?: no handler registered for (n1,s1):?
W200318 20:43:03.292145 400669 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: node unavailable; try another peer:
W200318 20:43:03.295943 400570 kv/kvserver/store.go:1541  [s3,r1/3:{/Min-a}] could not gossip first range descriptor: node unavailable; try another peer
I200318 20:43:03.348958 401285 util/stop/stopper.go:539  quiescing
W200318 20:43:03.357497 400457 gossip/gossip.go:1513  [n2] no incoming or outgoing connections
W200318 20:43:03.360330 400842 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200318 20:43:03.364061 400586 gossip/gossip.go:1513  [n3] no incoming or outgoing connections
W200318 20:43:03.365657 400624 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: EOF:
W200318 20:43:03.368753 401101 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 3: EOF:
W200318 20:43:03.371456 401074 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 1: EOF:
W200318 20:43:03.374391 401143 kv/kvserver/raft_transport.go:637  while processing outgoing Raft queue to node 2: rpc error: code = Unavailable desc = transport is closing:
I200318 20:43:03.388250 401285 util/stop/stopper.go:539  quiescing
I200318 20:43:03.390135 401285 util/stop/stopper.go:539  quiescing
I200318 20:43:03.396461 401285 util/stop/stopper.go:539  quiescing
    --- FAIL: TestReplicaRangefeedRetryErrors/REASON_RAFT_SNAPSHOT (4.03s)
        replica_rangefeed_test.go:590: incorrect events on stream, found [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> >  checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<wall_time:1584564182767312181 logical:1 > > ], want [checkpoint:<span:<key:"a" end_key:"z" > resolved_ts:<> > ]

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestReplicaRangefeedRetryErrors PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestReplicaRangefeedRetryErrors.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/replica_rangefeed_test.go
+++ b/pkg/kv/kvserver/replica_rangefeed_test.go
@@ -310,6 +310,7 @@ func TestReplicaRangefeedExpiringLeaseError(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
 	sc := kvserver.TestStoreConfig(nil)
+	sc.Clock = nil // manual clock
 	kvserver.RangefeedEnabled.Override(&sc.Settings.SV, true)
 	mtc := &multiTestContext{
 		storeConfig: &sc,
@@ -349,6 +350,7 @@ func TestReplicaRangefeedRetryErrors(t *testing.T) {
 	setup := func(subT *testing.T) (*multiTestContext, roachpb.RangeID) {
 		subT.Helper()
 		sc := kvserver.TestStoreConfig(nil)
+		sc.Clock = nil // manual clock
 		kvserver.RangefeedEnabled.Override(&sc.Settings.SV, true)
 		mtc := &multiTestContext{
 			storeConfig: &sc,

